easybuild/tools/toolchain/mpi.py

```diff
@@ -191,7 +191,8 @@ def mpi_cmd_for(self, cmd, nr_ranks):
             tmpdir = tempfile.mkdtemp(prefix='eb-mpi_cmd_for-')
 
             # set temporary dir for mdp
-            env.setvar('I_MPI_MPD_TMPDIR', tmpdir)
+            # note: this needs to be kept *short*, to avoid mpirun failing with "socket.error: AF_UNIX path too long"
+            env.setvar('I_MPI_MPD_TMPDIR', tempfile.gettempdir())
```

Comment:
how about something like:

```python
mpd_tmpdir_env_var = 'I_MPI_MPD_TMPDIR'
mpd_tmpdir_limit = 10
# set sensible value for $I_MPI_MPD_TMPDIR, unless it's set already
if os.getenv(mpd_tmpdir_env_var) is None:
  env.setvar(mpd_tmpdir_env_var, tempfile.gettempdir())
# check whether value for $I_MPI_MPD_TMPDIR isn't too long, to avoid problems
# e.g., mpdboot failing with "failed to ping mpd on localhost" due to "socket.error: AF_UNIX path too long"
if len(os.environ[mpd_tmpdir_env_var]) > mpd_tmpdir_limit:
    tup = (mpd_tmpdir_limit, os.environ[mpd_tmpdir_env_var])
    self.log.warning("Value for $%s must be (very) short to avoid problems, current value is longer than %d: %s" % tup)
```